diagnostic_rt_qpcr/operation_types/prepare_rt_qpcr_plate/protocol.rb

```diff
@@ -1,112 +1,188 @@
+
+  # Default parameters that are applied equally to all operations.
+  #   Can be overridden by:
+  #   * Associating a JSON-formatted list of key, value pairs to the `Plan`.
+  #   * Adding a JSON-formatted list of key, value pairs to an `Operation`
+  #     input of type JSON and named `Options`.
+  #
   def default_job_params
     {
-      primer_set: 'RP_N2',
-      max_inputs: 24,
-      group_size: 3,
-      method: 'sample_layout'.to_sym, 
-      additional_inputs: [
-        # {
-        #   name: nil,
-        #   container: nil,
-        #   station: 'bench'
-        #   volume{
-        #       qty: nil,
-        #       units: MIROLITERS
-        # }
-      ],
-      negative_controls: {
-        name: 'NTC',
-        first_well: [0, 0],
-        station: 'negative_control station',
-        volume: {
-          qty: 5,
-          units: MICROLITERS
-        },
-        additional_inputs: [
-          # {
-          #   name: nil,
-          #   container: nil,
-          #   station: nil,
-          #   volume{
-          #        qty: nil,
-          #        units: MIROLITERS
-          #   }
-          # }
-        ]
-      },
-      positive_controls:{
-        name: 'nCoVPC',
-        first_well: [0, 11],
-        station: 'positive control station',
-        volume: {
-          qty: 5,
-          units: MICROLITERS
-        },
-        additional_inputs: [
-          # {
-          #   name: nil,
-          #   container: nil,
-          #   station: nil,
-          #   volume{
-          #        qty: nil,
-          #        units: MIROLITERS
-          #   }
-          # }
-        ]
-      },
-      standard_samples: {
-        station: 'sample handling station',
-        volume: {
-          qty: 5,
-          units: MICROLITERS
-        },
-        additional_inputs: [
-          # {
-          #   name: nil,
-          #   container: nil,
-          #   station: nil,
-          #   volume{
-          #        qty: nil,
-          #        units: MIROLITERS
-          #   }
-          # }
-        ]
-      }
+      max_inputs: 24
     }
   end
 
+  # Default parameters that are applied to individual operations.
+  #   Can be overridden by:
+  #   * Adding a JSON-formatted list of key, value pairs to an `Operation`
+  #     input of type JSON and named `Options`.
+  #
   def default_operation_params
-    {}
+    {
+      negative_template_control: nil,
+      negative_template_location: nil,
+      positive_template_control: 'Test nCoVPC',
+      positive_template_location: [0, 11]
+    }
   end
 
   def main
-    operations.reject!{ |op| validate(operations: operations).include?(op) }
-    return {} if operations.empty?
 
-    operations.retrieve
+    #========= Setup Job ==========#
+    @job_params = update_all_params(
+      operations: operations,
+      default_job_params: default_job_params,
+      default_operation_params: default_operation_params
+    )
+
+    setup_test(operations) if debug
```

Comment:
This should happen before `update_all_params` (and anything else) because you need to be able to test all the code on what you set up, and not change the setup mid-stream.